rootchain/samuel/mock_test.go

```diff
@@ -0,0 +1,172 @@
+package samuel
+
+import (
+	"github.com/0xPolygon/polygon-edge/rootchain"
+	"github.com/0xPolygon/polygon-edge/rootchain/proto"
+)
+
+type startDelegate func(uint64) error
+type stopDelegate func() error
+type subscribeDelegate func() <-chan rootchain.Event
+
+type mockEventTracker struct {
+	startFn     startDelegate
+	stopFn      stopDelegate
+	subscribeFn subscribeDelegate
+}
+
+func (m mockEventTracker) Start(startBlock uint64) error {
+	if m.startFn != nil {
+		return m.startFn(startBlock)
+	}
+
+	return nil
```

Comment:
Is there any intended reason to check nil here? I think panic can tell us the existence of unexpected method call.

Reply:
This is how we do mock methods for 100% of mocks in the codebase. If it's not defined, I don't need it to panic

Reply:
In the first place, a test causing panic is missing something. If the mock has default behaviour, hard to notice unintentionally method call.

Reply:
Resolving this as it's irrelevant